logstash-core/src/main/java/org/logstash/ackedqueue/Page.java

```diff
@@ -54,16 +53,18 @@ public Batch readBatch(int limit) throws IOException {
         this.pageIO.activate();
 
         SequencedList<byte[]> serialized = this.pageIO.read(this.firstUnreadSeqNum, limit);
-        List<Queueable> deserialized = serialized.getElements().stream().map(e -> this.queue.deserialize(e)).collect(Collectors.toList());
-
+        List<byte[]> elements = serialized.getElements();
+        final int count = elements.size();
+        List<Queueable> deserialized = new ArrayList<>(count);
+        for (final byte[] element : elements) {
+            deserialized.add(this.queue.deserialize(element));
+        }
```

Comment:
Is this move away from stream/map/collect something you want to do in general, or just in this particular instance for specific performance reasons?